src/Symfony/Bundle/FrameworkBundle/Command/CacheClearCommand.php

```diff
@@ -79,17 +79,23 @@ protected function execute(InputInterface $input, OutputInterface $output): int
         $io = new SymfonyStyle($input, $output);
 
         $kernel = $this->getApplication()->getKernel();
+        $realBuildDir = $kernel->getContainer()->getParameter('kernel.build_dir');
         $realCacheDir = $kernel->getContainer()->getParameter('kernel.cache_dir');
         // the old cache dir name must not be longer than the real one to avoid exceeding
         // the maximum length of a directory or file path within it (esp. Windows MAX_PATH)
+        $oldBuildDir = substr($realBuildDir, 0, -1).('~' === substr($realBuildDir, -1) ? '+' : '~');
         $oldCacheDir = substr($realCacheDir, 0, -1).('~' === substr($realCacheDir, -1) ? '+' : '~');
-        $fs->remove($oldCacheDir);
+        $fs->remove([$oldBuildDir, $oldCacheDir]);
 
+        if (!is_writable($realBuildDir)) {
+            throw new RuntimeException(sprintf('Unable to write in the "%s" directory.', $realBuildDir));
+        }
         if (!is_writable($realCacheDir)) {
             throw new RuntimeException(sprintf('Unable to write in the "%s" directory.', $realCacheDir));
         }
 
         $io->comment(sprintf('Clearing the cache for the <info>%s</info> environment with debug <info>%s</info>', $kernel->getEnvironment(), var_export($kernel->isDebug(), true)));
+        $this->cacheClearer->clear($realBuildDir);
```

Comment:
Shouldn't this command avoid removing the build dir by default? What about adding a `--build` option?
For the ones not overriding `KernelInterface::getBuildDir()`, this option won't do anything more.
But for the ones overriding, you might want to call this command to clear the read-write cache, but not the build dir, right?

Reply:
As a first step, we must clear everything like today.